app/javascript/components/student/mentoring-session/mentoring-request/MentoringRequestForm.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useRef } from 'react'
+import {
+  CopyToClipboardButton,
+  FormButton,
+  GraphicalIcon,
+} from '../../../common'
+import { Track, Exercise } from '../../MentoringSession'
+import { useMutation } from 'react-query'
+import { sendRequest } from '../../../../utils/send-request'
+import { typecheck } from '../../../../utils/typecheck'
+import { useIsMounted } from 'use-is-mounted'
+import { MentoringRequest } from '../../../types'
+import { FetchingBoundary } from '../../../FetchingBoundary'
+
+type Links = {
+  learnMoreAboutPrivateMentoring: string
+  privateMentoring: string
+  mentoringGuide: string
+  createMentorRequest: string
+}
+
+const DEFAULT_ERROR = new Error('Unable to create mentor request')
+
+export const MentoringRequestForm = ({
+  isFirstTimeOnTrack,
+  track,
+  exercise,
+  links,
+  onSuccess,
+}: {
+  isFirstTimeOnTrack: boolean
+  track: Track
+  exercise: Exercise
+  links: Links
+  onSuccess: (mentorRequest: MentoringRequest) => void
+}): JSX.Element => {
+  const isMountedRef = useIsMounted()
+  const [mutation, { status, error }] = useMutation<
+    MentoringRequest | undefined
+  >(
+    () => {
+      return sendRequest({
+        endpoint: links.createMentorRequest,
+        method: 'POST',
+        body: JSON.stringify({ comment: solutionCommentRef.current?.value }),
+        isMountedRef: isMountedRef,
+      }).then((json) => {
+        if (!json) {
+          return
+        }
+
+        return typecheck<MentoringRequest>(json, 'mentorRequest')
+      })
+    },
+    {
+      onSuccess: (mentorRequest) => {
+        if (!mentorRequest) {
+          return
+        }
+
+        onSuccess(mentorRequest)
+      },
+    }
+  )
+
+  const trackCommentRef = useRef<HTMLTextAreaElement>(null)
+  const solutionCommentRef = useRef<HTMLTextAreaElement>(null)
+
+  return (
+    <div className="mentoring-request-section">
+      <div className="direct">
+        <h3>
+          Send this link to a friend for private mentoring.{' '}
+          <a href={links.learnMoreAboutPrivateMentoring}>Learn more</a>.
+        </h3>
+        <CopyToClipboardButton textToCopy={links.privateMentoring} />
+      </div>
+      <form className="community" onSubmit={() => mutation()}>
+        <div className="heading">
+          <div className="info">
+            <h2>It’s time to deepen your knowledge.</h2>
+            <p>
+              Start a mentoring discussion on <strong>{exercise.title}</strong>{' '}
+              to discover new and exciting ways to approach it. Expand and
+              deepen your knowledge.
+            </p>
+          </div>
+          <GraphicalIcon icon="graphic-mentoring-header" />
+        </div>
+        {isFirstTimeOnTrack ? (
+          <div className="question">
+            {/*TODO: @iHiD Instead of an h3, I made this a label */}
```

Comment:
Why? For testing? I think from an a11y POV it's better as a heading.

Reply:
You should be able to click the text and move focus on the textarea, since it's a form.

Reply:
Ok. I understand.

I'm going to go with h3 though. I think it's more correct from an a11y POV as the text above isn't really a label in my eyes, its more of a description. I've added aria-labelledby and aria-describedby tabs to the textareas to make it clear that they're linked.

Reply:
Urgh. I've gone back to labels as this breaks the tests, and gone with described by to make it clear the paragraph is linked to the input (as labels are normally supposed to go directly next to inputs)